File: src/main/java/uk/co/rafearnold/randomworkoutroutine/repository/GroupRepository.java
package uk.co.rafearnold.randomworkoutroutine.repository;

import uk.co.rafearnold.randomworkoutroutine.model.entity.ExerciseOption;
import uk.co.rafearnold.randomworkoutroutine.model.entity.Group;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface GroupRepository extends ItemRepository<Group> {

    List<Group> findAllByExerciseOptionsContaining(ExerciseOption exercise);
}

File: web/src/main/kotlin/uk/co/rafearnold/randomworkoutroutine/web/repository/GroupRepository.kt
package uk.co.rafearnold.randomworkoutroutine.web.repository

import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import uk.co.rafearnold.randomworkoutroutine.web.model.entity.ExerciseOption
import uk.co.rafearnold.randomworkoutroutine.web.model.entity.Group
import java.util.*

@Repository
interface GroupRepository : CrudRepository<Group, UUID> {
    fun findAllByExercisesContaining(exercise: ExerciseOption): List<Group>
}
